File: perl/regex/remove-md-bold.py
#!/usr/bin/env python3
import sys
import os
import re

if len(sys.argv) != 2:
    print(f"Usage: {sys.argv[0]} <markdown_file>", file=sys.stderr)
    sys.exit(1)

file = sys.argv[1]
if not os.path.exists(file):
    print(f"File '{file}' not found", file=sys.stderr)
    sys.exit(1)

# Read file
with open(file, 'r') as fh:
    content = fh.read()

# Remove bold formatting
content = re.sub(r'(\*\*|__)(.+?)\1', r'\2', content, flags=re.DOTALL)

# Write back to same file
with open(file, 'w') as fh:
    fh.write(content)

print(f"Removed bold formatting from {file}")

File: perl/regex/remove-md-bold.pl
#!/usr/bin/perl
use strict;
use warnings;

die "Usage: $0 <markdown_file>\n" unless @ARGV == 1;

my $file = $ARGV[0];
die "File '$file' not found\n" unless -e $file;

# Read file
local $/;
open( my $fh, '<', $file ) or die "Cannot read '$file': $!\n";
my $content = <$fh>;
close($fh);

# Remove bold formatting
$content =~ s/(\*\*|__)(.+?)\1/$2/gs;

# Write back to same file
open( $fh, '>', $file ) or die "Cannot write '$file': $!\n";
print $fh $content;
close($fh);

print "Removed bold formatting from $file\n";
